tests/e2e/tests/test_cancellation_v1.go

```diff
@@ -0,0 +1,598 @@
+	ctx context.Context,
+	t *testing.T,
+	tt *Test,
+	supplierPPEventStream events.EventsService_SubscribeClient,
+	distributorBot *bot.Bot,
+	supplierBot *bot.Bot,
+) (uint64, *typesv3.Price) {
+	searchID, resultID, totalPrice := testAccommodationV3SearchServiceWithTravelPeriod(ctx, t, tt, distributorBot, supplierBot) // see test_accommodation_v3.go
+	_, err := supplierPPEventStream.Recv()                                                                                      // skip AccommodationSearchRequest
+	require.NoError(t, err)
+
+	validationID := testAccommodationV3ValidateV2(ctx, t, tt, distributorBot, supplierBot, searchID, resultID, totalPrice) // see test_accommodation_v3.go
+	_, err = supplierPPEventStream.Recv()                                                                                  // skip ValidateRequest
+	require.NoError(t, err)
+
+	tokenID, bookingPrice, _ := testAccommodationV3MintV3(ctx, t, tt, distributorBot, supplierBot, validationID) // see test_accommodation_v3.go
+	_, err = supplierPPEventStream.Recv()                                                                        // skip MintRequest
+	require.NoError(t, err)
+
+	eventMsg, err := supplierPPEventStream.Recv()
+	require.NoError(t, err)
+	debugPrintProtoMessage(tt, eventMsg)
+	tokenBoughtNotification := &notificationv2.TokenBought{}
+	require.NoError(t, proto.Unmarshal(eventMsg.Data, tokenBoughtNotification))
+
+	return tokenID, bookingPrice
+}
+
+func testAccommodationV3MintV3(
+	ctx context.Context,
+	t *testing.T,
+	tt *Test,
+	distributorBot *bot.Bot,
+	supplierBot *bot.Bot,
+	validationID string,
+) (
+	tokenID uint64,
+	price *typesv3.Price,
+	mintID string,
+) {
+	req := &bookv3.MintRequest{
+		Header:       &typesv1.RequestHeader{BaseHeader: &typesv1.Header{}},
+		ValidationId: &typesv1.UUID{Value: validationID},
+	}
+	resp, err := distributorBot.MintServiceV3.Mint(
+		requestContext(ctx, &metadata.Metadata{
+			RecipientCMAccount: supplierBot.CMAccountAddress().Hex(),
+		}),
+		req,
+	)
+	require.NoError(t, err)
+	debugPrintRequestResponse(tt, getCurrentFuncName(), req, resp)
+
+	require.Equal(t, typesv1.StatusType_STATUS_TYPE_SUCCESS, resp.Header.Status, "unexpected response status")
+
+	// Check if the MintId is set
+	require.NotEmpty(t, resp.MintId, "unexpected empty response MintId")
+	require.NotEmpty(t, resp.MintId.Value, "unexpected empty response MintId.Value")
+
+	// check if the transaction ids are set and return them for further tests
+	require.NotEmpty(t, resp.MintTransactionId, "unexpected empty response MintTransactionId")
+	require.NotEmpty(t, resp.BuyTransactionId, "unexpected empty response BuyTransactionId")
+
+	return resp.BookingTokenId, resp.Price, resp.MintId.Value
+}
+
+func testCancellationV1(
+	ctx context.Context,
+	t *testing.T,
+	tt *Test,
+	distributorPPEventStream events.EventsService_SubscribeClient,
+	supplierPPEventStream events.EventsService_SubscribeClient,
+	distributorBot *bot.Bot,
+	supplierBot *bot.Bot,
+) {
```

Comment:
Good coverage of different use cases. I think the most common one is missing which would be a simple:
Distributor initiates -> Supplier accepts (-> Distributor finalizes?)

Reply:
it should be then "Distributor initiates -> Supplier Finalizes"
supplier always finalize, because he owns the funds